tests/unit/autoprompt/test_widgets.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import mock
+
+from prompt_toolkit.formatted_text import FormattedText
+from prompt_toolkit.layout import HSplit, Window, VSplit, ConditionalContainer
+
+from awscli.autoprompt.widgets import (
+    FormatTextProcessor, ToolbarWidget, HelpPanelWidget
+)
+
+from awscli.testutils import unittest
+
+
+class TestToolbarWidget(unittest.TestCase):
+    def setUp(self):
+        self.widget = ToolbarWidget()
+
+    def test_can_create_container(self):
+        self.assertTrue(isinstance(self.widget.container, HSplit))
+
+    def test_container_has_correct_children(self):
+        children = self.widget.container.get_children()
+        self.assertTrue(isinstance(children[0], ConditionalContainer))
+        self.assertTrue(isinstance(children[1], VSplit))
+
+    def test_toolbar_history_wording(self):
+        buffer = self.widget.container.get_children()[1]\
+                            .get_children()[0].content.content.buffer
+        self.assertEqual(buffer.document.text, self.widget.history_mode_sign())
```

Comment:
I think if we break the methods out into separate classes, I think that will improve the tests as well. Right now it is a little hard to understand with the children traversal and assertions of general classes what we are trying to test.